app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# :nodoc:
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :sign_post, expect: %i[new index]
+
+  def new
+    @post = Post.new
+  end
+
+  def index
+    @posts = Post.all
+  end
+
+  def show; end
+
+  def create
+    @post = current_user.post.build(permit_post)
+    if @post.save
+      flash[:success] = 'The Post you created was a Success!'
+      redirect_to posts_path(@post)
+    else
+      flash[:error] = @post.errors.full_messages
+      redirect_to new_post_path
```

Comment:
redirect is na new request we don't need. let use render to just bring the page that need to be displayed instead. and use flash.now[:error] so the flash can still pass.

Reply:
submitting a blank form return error as expected 👍  lets present it in a better way without the array thing:
![Screenshot 2019-09-26 at 2 15 03 PM](https://user-images.githubusercontent.com/10922508/65691091-2b593d00-e068-11e9-9a14-0fbd2654f279.png)
- when a new post is created, the notification is at the bottom! NOOOO. look at the flash code in `application.html.erb`

Reply:
lol. yeah flash code was placed above the yield action in application.htmk.erb